### ISSUE TITLE
Service immediately fails

### ISSUE THREAD

**Issue by averyhox** [TapeWerm/MCscripts#5]
Running Ubuntu 18.04 Server
Minecraft Bedrock Edition 1.11.2.1
Running the server for a few months now without MCscripts

Likely an issue on my end.  I am going to blow my minecraft instance entirely and start over and hopefully it'll fix it.  However, while I am working through that, I wanted to see if any of these issues are familiar or if you could clarify some comments/commands.

_____________________________________
I followed the instructions on the description, but I believe I missed some instructions.  For example, the basic 'how do you start the server' seems lost to me.  This may be why I am running into a few issues.

1. Followed common setup, no issues and everything successfully installed 
2. Transitioned to Bedrock Edition Setup and all steps worked properly

At this point, where is the 'go' button?  As the server files were moved, new account was created, and tmux added, I guess opening the server directly through LD_LIBRARYPATH=. ./bedrock_server is no longer the proper way to open the server.  I assume the notes section is where I am to go.

1.  sudo su mc -s /bin/bash   - no issues
2. tmux -S /tmp/tmux-mc/minecraft a    - returns no session

Questions:  Is $instance any name I want to give it?  Is minecraft not an option?  Does it need to be something in particular?

I then followed the closed issue and it discussed tmux -S /tmp/MCBE new -ds MCBE 'LD_LIBRARY_PATH=. ./bedrock_server'

It seems this was a better command and was sending me in the right direction.  I could use an amalgamation of this command and get the server to start up.  However, I believe due to the previous commands, it would shut the server down due to the port being in use.  Although, after checking all ports, it was in fact not in use.  Not sure why this issue occurred.  After reboot and running the server directly from the folder with LD_LIBRARY_PATH=. ./bedrock_server, it continues to toss the "port in use" error. 

Ultimately the question I have is which command is the proper command -- the first command from the Description page or the second command on the issues page.  I assume the /tmp directory differences does not matter, but it seems the first command includes a /$instance (not quite sure what this is for) and the second command ignores this altogether and instead creates a new session and calls the server directly.

Additionally, if I just run the server the normal way, will the systemctl services I set up earlier auto run and continue to update and backup the server without the tmux session?  Or is the way the new user mc is set up that the systemctl sessions will only work if the minecraft server is started from the mc user?

Thanks for any information you provide.  Good work on the scripts... will be a massive time saver when they are up and running.  Bedrock updates are a slight annoyance.

**Comment by averyhox** [comment]
Quick update.

After a few reboots, re-configure, etc the minecraft server starts on boot automatically.   

Most of my questions remain for clarity if/when you get a chance.  

Primarily:  Will the systemctl scripts continue to run although the tmux command was not ran under the MC user?
     -  If I run sudo systemctl enable mcbe@MCBE.server --now 

> created symlink /etc/systemd/system/multi-user.target.wants/mcbe@MCBE.server.service -> /etc/systemd/system/mcbe@.service.
> Job for mcbe@MCBE.server.service failed because the control process exited with error code.

**Comment by TapeWerm** [comment]
Enabling the mcbe service instance MCBE (AKA mcbe@MCBE) with the --now flag starts the service. `sudo service mcbe@MCBE start` would be your go button if you stopped it or it crashed. Launching it the old way (`LD_LIBRARYPATH=. ./bedrock_server`) would technically work, but the service wouldn't be considered on. The backup service depends on the mcbe service, and wouldn't start. The service also has sandboxing, if you open a new pane in the service's tmux session you'll notice many dirs normally readable by the mc user aren't readable. `tmux -S /tmp/tmux-mc/minecraft a` should work if the instance of the service is named minecraft (mcbe@minecraft). You can give the instance any name, I think symbols/control characters/emoji may make systemd angry but I never tried being a bad person so I cannot verify that.
@@@
Do not follow that advice I gave in the closed issue, I made a new directory in `/tmp` called `/tmp/tmux-mc`. You would need to update the systemd unit files for mcbe and mcbe-backup for `/tmp/$instance` to be the right path, otherwise it would be fine. `tmux -S $path` is the socket the tmux session uses, and only one service can use the same socket. As mentioned, the backup service needs the mcbe service, but the update service does not as it will stop the service anyways. Thus, I decided to have no dependency, though it and the backup service are conflicts with each other as one needs the mcbe service up and one needs it down.

Also, you typed MCBE.server, you're supposed to type `sudo systemctl enable mcbe@MCBE.service --now`. Does the port thing still happen? Someone talked with me on Discord about it and could not solve their issue.

**Comment by TapeWerm** [comment]
Also, try
`service mcbe@MCBE status`
`service mcbe-backup@MCBE status`
`service mcbe-autoupdate@MCBE status`

I also recommend `git pull origin master` and copying the *.sh files again, you'll see in `git log` that I mention changing the flags and globbing. There was a bug where the autoupdate script would fail because it tried to remove old ZIPs when there wasn't any. Ironically it would succeed at updating the server as the new bedrock server ZIP was downloaded and the regular update script ran successfully. Just something good to update. The globbing fix is just good practice, it now includes dotfiles but bedrock server dirs have no dotfiles anyways.

**Comment by TapeWerm** [comment]
Another good command is `journalctl -u mcbe@MCBE`, that will show way, way more logs than service status will. Press the End key or Shift-G to go to the most recent logs, journalctl will start at the top.

**Comment by averyhox** [comment]
> sudo systemctl enable mcbe@MCBE.service --now

Worked without issue

> service mcbe@MCBE status
> service mcbe-backup@MCBE status
> service mcbe-autoupdate@MCBE status

All of the service checks work.... and I guess something I did last night started it up, as it successfully created a backup around 4am and checked for an update around midnight.

Thanks for the additional information in your reply.  They give me the exact information I need to restart fresh to ensure I did not muck it up the first time.
